server/src/routes/api/v1/squidsRouter.js

```diff
@@ -8,8 +8,17 @@ export const squidsRouter = new express.Router();
 squidsRouter.get(
   "/",
   nextWrapper(async (req, res) => {
-    const squids = await Squid.query();
+    const squidsQuery = Squid.query();
+    if (req.query.offset) {
+      const [totalSquidCount, squids] = await Promise.all([
+        squidsQuery.resultSize(),
+        squidsQuery.offset(req.query.offset).limit(req.query.limit),
+      ]);
 
-    return res.status(200).json(squids);
+      const pageCount = Math.ceil(totalSquidCount / req.query.limit);
+      return res.status(200).json({ squids, pageCount });
+    }
+    const squids = await squidsQuery;
```

Comment:
Here, we should enforce pagination, even if the request doesn't include an offset -- if you had a bazillion squids, someone could spam this endpoint to overload your server and bring down your app 💥 (checking in with Dan on the term for that, will follow up once I know!)

Reply:
oh, or -- you could just run out of memory regardless of whether you have a user with malicious intent :) One query that returns millions would do that